heggy-heggy231/week4/express-intro-lab-master/server.js

```diff
@@ -0,0 +1,35 @@
+// server.js
+// resource about express: http://expressjs.com/en/4x/api.html#app
+// intro lab: https://git.generalassemb.ly/sf-wdi-51/express-intro-lab
+// SERVER-SIDE JAVASCRIPT
+
+// export express library
+const express = require('express');
+// app receives top level express function exported by express module
+// The app object has methods for
+// Routing HTTP requests; see for example, app.METHOD and app.param.
+// Configuring middleware; see app.route.
+// Rendering HTML views; see app.render.
+// Registering a template engine; see app.engine.
+const app = express();
+
+// Allow CORS: we'll use this today to reduce security so we can more easily test our code in the browser.
+//  next: Callback argument to the middleware function
+app.use((req, res, next) => {
+  res.header("Access-Control-Allow-Origin", "*");
+  res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+  next();
+});
+
```

Comment:
Good job breaking it down step by step. This will help your understanding of the bigger picture of how node and express works together.